README.md

```diff
@@ -38,6 +38,72 @@ If you want to test what happens when the HTTP server goes down, use `Bypass.dow
 TCP socket and `Bypass.up/1` to start listening on the same port again. Both functions block until
 the socket updates its state.
 
+### Expect Functions
+
+You can take any of the following approaches:
+* `expect/2` or `expect_once/2` to install a generic function that all calls to bypass will use
+* `expect/4` and/or `expect_once/4` to install specific routes (method, path combinations)
+* a combination of the above, where the routes will be used first, and then the generic version
+  will be used as default
+
+#### expect/2 (bypass_instance, function)
+
+Must be called at least once, and it will error if not.  As an alternative, `nil` can be passed
```

Comment:
Even though supporting `nil` is already part of Bypass, I would still like to phase it out for the reasons I mentioned above. It is usually better to start strict and relax some constraints if they prove to be too stringent in practice.

The test case you've mentioned doesn't look very useful in practice. I think pretty much everyone would benefit if Bypass could be used in a more declarative way. That's also the reason for adding functions like `expect_once` and others. Most of them could be implemented with a single `expect()` function that uses some external state but that would make the test code much less clear.

I don't see staying backwards compatible as important here. This PR is already introducing pretty significant additions, so it'll probably get released as a new major version. Cleaning up dubious existing behaviour would be part of the major release.